pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -454,7 +464,12 @@ func (r *ReconcilePerconaServerMongoDB) reconcileStatefulSet(arbiter bool, cr *a
 	if err != nil {
 		return nil, fmt.Errorf("create StatefulSet.Spec %s: %v", sfs.Name, err)
 	}
-
+	if sfsSpec.Template.Annotations == nil {
+		sfsSpec.Template.Annotations = make(map[string]string)
+	}
+	for k, v := range sfs.Spec.Template.Annotations {
+		sfsSpec.Template.Annotations[k] = v
+	}
```

Comment:
I don't quite like an idea of updating sfs in two places. It's kinda race. There is no guarantee that by time reading sfs here https://github.com/percona/percona-server-mongodb-operator/pull/387/files#diff-0e749e0e5eac1be1dc8fd10a7b2a0a3aR463 you've got all the changes from here https://github.com/percona/percona-server-mongodb-operator/pull/387/files#diff-49949a3da6b9086a89870164ade89c83R234 fully applied

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/387/commits/fea60f14bf2ff9b7f74adf4d600e967978d6c92a